features/kinderen.feature

```diff
@@ -1,71 +1,73 @@
+      En bevat het antwoord GEEN kind met naam.voornamen "Celeke Lodivica"
+  
 
-  Scenario: het kind is geen ingeschreven persoon
-    Gegeven de te raadplegen persoon heeft een kind die zelf geen ingeschreven persoon is
-    En het kind van de ingeschreven persoon heeft volgens categorie 09/59 naam Chantal Groenen, geboren in Istanbul en geboortedatum 2 januari 2002
-    Als de kinderen worden geraadpleegd van de ingeschreven persoon met burgerservicenummer 999999291
-    Dan wordt het kind gevonden met naam.voornamen=Chantal
-    En heeft dit kind burgerservicenummer=null
-    En heeft dit kind naam.voorvoegsel=null
-    En heeft dit kind naam.geslachtsnaam=Groenen
-    En heeft dit kind naam.adellijkeTitel_predikaat.omschrijvingAdellijkeTitel_predikaat=null
-    En heeft dit kind geboorte.datum.datum=2002-01-02
-    En heeft dit kind geboorte.datum.dag=02
-    En heeft dit kind geboorte.datum.maand=01
-    En heeft dit kind geboorte.plaats=Istanbul
-    En heeft dit kind geboorte.land.landnaam=Turkije
-    En heeft dit gevonden kind een lege link ingeschrevenpersonen
+  Rule: Een kind wordt alleen teruggegeven als minimaal één gegeven in de identificatienummers (groep 01), naam (groep 02) of geboorte (groep 03) van het kind een waarde heeft.
+    # wanneer in een categorie kind alleen gegevens zijn opgenomen in groep 81 of 82, 85 en 86, wordt dit kind niet opgenomen in het antwoord
 
-  Scenario: de ingeschreven persoon twee kinderen
-    Gegeven de te raadplegen persoon heeft meerdere (twee) kinderen (Lisa en Daan)
-    Als de kinderen worden geraadpleegd van de ingeschreven persoon met burgerservicenummer 999999047
-    Dan is het aantal gevonden kinderen 2
-    En wordt het kind gevonden met naam.voornamen=Lisa
-    En wordt het kind gevonden met naam.voornamen=Daan
+    Scenario: kind volledig onbekend
+      Gegeven de persoon met burgerservienummer 555550001 heeft de volgende kinderen in de registratie
+        | Categorie | Voornamen (02.10)  | Voorvoegsel (02.30) | Geslachtsnaam (02.40) | Geboortedatum (03.10) |
+        | 9         |                    |                     | .                     |                       |
+      En er zijn geen andere gegevens geregistreerd over de identificatienummers (groep 01), naam (groep 02) of geboorte (groep 03) in categorie 9
+      Als de ingeschreven persoon met burgerservicenummer 555550001 wordt geraadpleegd met fields=kinderen
+      Dan bevat het antwoord GEEN kinderen
 
-  Scenario: de ingeschreven persoon heeft drie kinderen
-    Gegeven de te raadplegen persoon heeft drie kinderen (Jip, Chantal en Jesse)
-    Als de kinderen worden geraadpleegd van de ingeschreven persoon met burgerservicenummer 999999291
-    Dan is het aantal gevonden kinderen 3
-    En wordt het kind gevonden met naam.voornamen=Jip
-    En wordt het kind gevonden met naam.voornamen=Chantal
-    En wordt het kind gevonden met naam.voornamen=Jesse
+    Scenario: ontkenning ouderschap
```

Comment:
Is deze scenario relevant? De rule zegt dat een kind niet wordt getoond bij een persoon als er voor het kind geen gegevens zijn geregistreerd voor groep01, groep02 en groep03. Dan maakt het toch niet uit of 'ouderschap is ontkend' of dat er een andere reden is dat het kind niet mag worden getoond bij de persoon?

Reply:
het scenario (gegeven) was niet helemaal compleet. het eerste scenario, dat niks bekend is, is anders, want die zegt dat er een kind is, maar we niet weten hoe die heet. We tonen die niet, omdat er niks te tonen is.
Het tweede scenario, gaat over de situatie dat er eerst een kind geregistreerd stond, maar dat die is verwijderd van de PL, omdat de ouder het ouderschap ontkent. Dus er is wel een naam van het kind bekend, maar dit blijkt geen kind van de persoon te zijn.
Technisch voor de implementatie van de provider zijn beide scenario's wellicht gelijk, maar voor een gebruiker zijn ze denk ik wel relevant.

Reply:
ik heb de redenatie van issue #214 gevolgd, specifiek de uitspraak van @CathyDingemanse in https://github.com/VNG-Realisatie/Haal-Centraal-BRP-bevragen/issues/214#issuecomment-496428770: "de gegevens van gerelateerden die er niet zijn of onjuist zijn of onbekend zijn ("." of "00000000") niet te leveren. M.a.w. alleen iets leveren als er ook echt iets te melden is.". Die beslissing stel je nu ter discussie.

Dit is denk ik een nogal exotische situatie, ik kan me geen situatie bedenken dat dit voor zal komen (bij ouders komt dat wel regelmatig voor), maar stel dat geregistreerd is dat er een kind is, maar totaal onbekend is hoe die heet en wanneer die geboren is, moet de gebruiker dan alsnog op de een of andere manier doorkrijgen dat er wel een kind is?

Reply:
> 
 
Na de diverse discussies van de afgelopen tijd wel. Mag/kan de API gegevens weglaten die zijn geregistreerd. Kan de consumer van de API door het niet verstrekken van informatie niet verkeerde conclusies trekken?

Reply:
@CathyDingemanse gaat na of het consequenties kan hebben voor gemeentelijke processen dat er een kind is, ook wanneer er niks van dat kind bekend is (naam en geboortedatum). Idem voor partner.